tiatoolbox/tools/patchextraction.py

```diff
@@ -60,12 +60,22 @@ class PatchExtractor(ABC):
 
     """
 
-    def __init__(self, input_img, patch_size, resolution=0, units="level"):
+    def __init__(
+        self,
+        input_img,
+        patch_size,
+        resolution=0,
+        units="level",
+        pad_mode="constant",
+        pad_constant_values=0,
```

Comment:
What are the options for `pad_mode`? Is it similar to np.pad ?

Reply:
I think this may be within Shan PR for patch extract, is this after you pulled that one over? I will skip over the PatchExtractor for now.

Reply:
> What are the options for `pad_mode`? Is it similar to np.pad ?

Exactly the same, also see read_bounds and read_rect which links to np.pad. This could be added here but I am aware that Shan is reworking this also and might be re-arranged.

Reply:
https://github.com/TIA-Lab/tiatoolbox/pull/61/commits/3a4f5f2b6ecd33bcf499f19e109fafbef44b0e94